### PR TITLE
Mol 13278

### PR DESCRIPTION
**Pull request by Huizz**
**Changes**
Description of changes...

-  delete branch
- Update default maxAttempts to 3 for getCurrentLocation
- Only call Error.captureStackTrace if Error.captureStackTrace is available

**Changelog entry**

- Fixes on location-input component

### CODE REVIEW

File: src/services/onemap/types.ts [hunk 1]
```diff
@@ -60,6 +60,8 @@ export class OneMapError extends Error {
 		super(message);
 		this.name = "OneMapError";
 		this.innerError = error;
-		Error.captureStackTrace(this, OneMapError);
+		if (Error.captureStackTrace) {
+			Error.captureStackTrace(this, OneMapError);
+		}
```
Comment (weili-govtech) [inline src/services/onemap/types.ts:65]: can use optional chaining

```suggestion
Error.captureStackTrace?.(this, OneMapError);
```

File: src/utils/geolocation-helper.ts [hunk 2]
```diff
@@ -23,15 +23,14 @@ export namespace GeoLocationHelper {
...
-	 * and whether to disable the error prompt in the LifeSG app (default: false).
+	 * the timeout in milliseconds (default: 2000), the maximum age of a possible cached position in milliseconds (default: 2000)
 	 * A minimum of one attempt will be made, even if maxAttempts is 0.
 	 * @returns A Promise that gets the current location.
 	 * @resolve type: ILocationCoord
 	 * @reject type: GeolocationPositionError | undefined
 	 */
 	export const getCurrentLocation = async (options: IGetCurrentLocationOptions = {}): Promise<ILocationCoord> => {
-		const { maxAttempts = 1, timeout = 2000, maximumAge = 2000 } = options;
+		const { maxAttempts = 3, timeout = 2000, maximumAge = 2000 } = options;
```
Comment (weili-govtech) [inline src/utils/geolocation-helper.ts:33]: what's the rationale of changing it? I think the original intention was to offer retry as an option

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/utils/geolocation-helper.ts (hunk 2 above)
Comment (Huizz) [reply src/utils/geolocation-helper.ts:33]: oh I asked Hilman and he mentioned the default should be 3 instead so I updated it.

Retry is also not offered as an option now though. so I should be exposing that instead? what's the previous discussion on this?

Comment (weili-govtech) [reply src/utils/geolocation-helper.ts:33]: ok I have checked with Hilman, it appears it's more ideal to have retry enabled by default. (when maxAttempts=1, it's akin to no retry enabled)

however can we update the comments to reflect the new behaviour
```ts
/**
 * @param options An optional options object that lets you specify the maximum no. of attempts (default: 3),
 * the timeout in milliseconds (default: 2000), the maximum age of a possible cached position in milliseconds (default: 2000),
 * A minimum of one attempt will be made, even if maxAttempts is 0.
 * @returns A Promise that gets the current location.
 * @resolve type: ILocationCoord
 * @reject type: GeolocationPositionError | undefined
*/
```

File: src/components/fields/location-field/location-modal/types.ts [hunk 3]
```diff
@@ -12,6 +12,7 @@ export interface ILocationModalProps
 	className: string;
 	showLocationModal: boolean;
 	formValues?: ILocationFieldValues | undefined;
+	locationModalStyles?: string;
```
Comment (weili-govtech) [inline src/components/fields/location-field/location-modal/types.ts:15]:
```suggestion
	locationModalStyles?: string | undefined;
```

File: src/components/fields/location-field/types.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ export interface ILocationFieldSchema<V = undefined>
 		Pick<ILocationInputProps, "locationInputPlaceholder">,
 		Pick<IStaticMapProps, "staticMapPinColor"> {
 	className?: string;
+	locationModalStyles?: string;
```
Comment (weili-govtech) [inline src/components/fields/location-field/types.ts:17]:
```suggestion
	locationModalStyles?: string | undefined;
```

**Review by weili-govtech — CHANGES_REQUESTED** [review]

**Review by weili-govtech — APPROVED** [review]

**Pull request merged**